### DISCUSSION TITLE
Limit the text in table column with the help of limit() and words() function.

### DISCUSSION THREAD

**Discussion by devRonak** [filamentphp/filament#6988] [Help]
I am trying to limit the table content with the limit function and words function, But also I used an `html()` function with that. it is not worked with `limit(50)` and `words(10)` not worked with `html()`. Does anyone know how can i achieve this with `html()` function?

My code:- 

`TextColumn::make('change_log')->words('10')->html()`

**Comment by pboivin** [comment]
I think `limit()` `words()` and `html()` are not compatible with each other (they're all presets for `formatStateUsing`). Have a look, maybe you can make your own combination:
https://github.com/filamentphp/filament/blob/2.x/packages/tables/src/Columns/Concerns/CanFormatState.php#L78

**Reply by devRonak** [reply]
Yes you are right @pboivin 
I tried to do some code but did not get the desired result. I don't know how I can achieve this.

**Reply by pboivin** [reply]
Can you share what you have tried?

**Reply by devRonak** [reply]
```
TextColumn::make('change_log')->render(function ($value) {
    $truncatedValue = Str::limit($value, 10);
    return "<span title='{$value}'>{$truncatedValue}</span>";
})
```

**Reply by devRonak** [reply]
I got the below answer from Discord.

=>Because it could truncate the text causing the cutoff of closing html tags which would result in breaking the entire page.

**Reply by pboivin** [reply]
Try this :
```
TextColumn::make('change_log')
    ->formatStateUsing(function ($state) {
        $truncatedValue = Str::limit($state, 10);
        return new HtmlString("<span title='{$state}'>{$truncatedValue}</span>");
    })
```

**Reply by devRonak** [reply]
Hey, @pboivin 
Thank you for your response.
Right now the code doesn't work 100%. The code currently only works when there is no HTML in the text.
For example when I used `<ul></ul>`  that time this code is not worked for me.
Thank You

**Reply by nickwild-999** [reply]
FYI. This works for me with limit()

**Comment by egekibar** [comment]
use only

```php
->limit(10)
```

**Comment by boumaane** [comment]
Is there a way to apply it on the description as well ?

```
TextColumn::make('category')
                    ->description(fn (Ticket $record) => new HtmlString($record->issue))
                    ->limit(30)
                    ->lineClamp(2),
```

**Reply by boumaane** [reply]
in the same way suggested above :

```
TextColumn::make('category')
                    ->description(function (Ticket $record){ 
                        $truncatedValue = Str::limit($record->issue, 30);
                        return new HtmlString("<span title='{$record->issue}'>{$truncatedValue}</span>");
                    })
                    ->lineClamp(2),
```
